src/app/copy/copy-base.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  EventEmitter,
+  Input,
+  Output,
+} from '@angular/core';
+
+import { CopyService } from './copy-service/copy.service';
+
+export abstract class CopyBase {
+  @Input('copyBtnAriaLabel') copyBtnAriaLabel: string;
+  @Input('copyValue') copyValue: string = 'Missing \'copyValue\' @Input property';
+  @Input('tooltip') tooltip: string;
+  @Input('tooltipPlacement') tooltipPlacement: string = 'top';
+
+  /**
+   * Event emitted with the chart reference after load is complete
+   * @type {EventEmitter}
+   */
+  @Output('copiedToClipboard') copiedToClipboard: EventEmitter<any> = new EventEmitter();
+
+  private _recentlyCopied: boolean = false;
+
+  /**
+   * Returns the flag indicating copy action has just happened
+   *
+   * @returns {boolean} True if copy action has been triggered
+   */
+  public get recentlyCopied(): boolean {
+    return this._recentlyCopied;
+  }
+
+  /**
+   * Set the flag indicating copy action has just happened
+   *
+   * @param {boolean} copied True when copy action has been triggered
+   */
+  public set recentlyCopied(copied: boolean) {
```

Comment:
Does this setter need to be public? I'm not sure it's used by the component itself, so perhaps it can be removed?

Reply:
Similar to copyValueToClipboard, if I change this to private/protected I get build errors. It's possible I'm just missing something so I'll take another look.

Reply:
Oh! Geez I feel silly now :) For some reason I was thinking the getters and setters were a package deal and I needed to _use_ the setter. Was definitely overcomplicating this, good call!